apstools/plans/xpcs_mesh.py

```diff
@@ -0,0 +1,237 @@
+
+
+def mesh_list_grid_scan(detectors, *args, number_of_collection_points, snake_axes=False, per_step=None, md=None):
+    """
+    Scan over a mesh; each motor is on an independent trajectory.
+
+    Parameters
+    ----------
+    detectors: list
+        list of 'readable' objects
+    args: list
+        patterned like (``motor1, position_list1,``
+                        ``motor2, position_list2,``
+                        ``motor3, position_list3,``
+                        ``...,``
+                        ``motorN, position_listN``)
+
+        The first motor is the "slowest", the outer loop. ``position_list``'s
+        are lists of positions, all lists must have the same length. Motors
+        can be any 'settable' object (motor, temp controller, etc.).
+    number_of_collection_points: int
+        The total number of collection points that must be collected within the
+        grid until the scan is ready to stop.
+    snake_axes: boolean or iterable, optional
+        which axes should be snaked, either ``False`` (do not snake any axes),
+        ``True`` (snake all axes) or a list of axes to snake. "Snaking" an axis
+        is defined as following snake-like, winding trajectory instead of a
+        simple left-to-right trajectory.The elements of the list are motors
+        that are listed in `args`. The list must not contain the slowest
+        (first) motor, since it can't be snaked.
+    per_step: callable, optional
+        hook for customizing action of inner loop (messages per step).
+        See docstring of :func:`bluesky.plan_stubs.one_nd_step` (the default)
+        for details.
+    md: dict, optional
+        metadata
+
+    See Also
+    --------
+    :func:`bluesky.plans.rel_list_grid_scan`
+    :func:`bluesky.plans.list_scan`
+    :func:`bluesky.plans.rel_list_scan`
+    """
+
+    full_cycler = plan_patterns.outer_list_product(args, snake_axes)
+
+    md_args = []
+    motor_names = []
+    motors = []
+    for i, (motor, pos_list) in enumerate(partition(2, args)):
+        md_args.extend([repr(motor), pos_list])
+        motor_names.append(motor.name)
+        motors.append(motor)
+        # print("when")
+    _md = {
+        "shape": tuple(len(pos_list) for motor, pos_list in partition(2, args)),
+        "extents": tuple([min(pos_list), max(pos_list)] for motor, pos_list in partition(2, args)),
+        "snake_axes": snake_axes,
+        "plan_args": {"detectors": list(map(repr, detectors)), "args": md_args, "per_step": repr(per_step)},
+        "plan_name": "list_grid_scan",
+        "plan_pattern": "outer_list_product",
+        "plan_pattern_args": dict(args=md_args, snake_axes=snake_axes),
+        "plan_pattern_module": plan_patterns.__name__,
+        "motors": tuple(motor_names),
+        "hints": {},
+    }
+    _md.update(md or {})
+    try:
+        _md["hints"].setdefault("dimensions", [(m.hints["fields"], "primary") for m in motors])
+    except (AttributeError, KeyError):
+        ...
```

Comment:
It would be useful/informative to call `logger.warning()` for these exceptions.  Capture the exception reason and report it to the logger:  `except (AttributeError, KeyError) as reason:`

Reply:
As copied from bluesy list_grid_scan, can change it but that is the way it is currently standardized. Please advise.